src/model/range.js

```diff
@@ -479,102 +469,58 @@ export default class Range {
-				// so this formula might look overcomplicated.
-				// However in some scenarios, after operational transformation, move operation might not
-				// in fact start from 0 and we need to properly count new offset.
-				// https://github.com/ckeditor/ckeditor5-engine/pull/1133#issuecomment-329080668.
-				const offset = this.start.offset - sourceRange.start.offset;
-
-				return [ new Range( targetPosition.getShiftedBy( offset ) ) ];
-			}
-			//
-			// Edge case for split delta.
-			//
-			if ( deltaType == 'split' && this.isCollapsed && this.end.isEqual( sourceRange.end ) ) {
-				// Collapsed range is at the end of split element.
-				// Without fix, the range would end up at the end of split (old) element instead of at the end of new element.
-				// That would happen because this range is not technically inside moved range. Last step below shows the fix.
-				// <p>foobar[]</p> -> <p>foobar[]</p><p></p> -> <p>foo[]</p><p>bar</p> -> <p>foo</p><p>bar[]</p>
-				return [ new Range( targetPosition.getShiftedBy( howMany ) ) ];
-			}
-			//
-			// Other edge cases:
+	_getTransformedByMoveOperation( operation, spread = false ) {
+		const sourcePosition = operation.sourcePosition;
+		const howMany = operation.howMany;
+		const targetPosition = operation.targetPosition;
+
+		return this._getTransformedByMove( sourcePosition, targetPosition, howMany, spread );
+	}
+
+	_getTransformedBySplitOperation( operation ) {
+		const start = this.start._getTransformedBySplitOperation( operation );
+		const end = this.end._getTransformedBySplitOperation( operation );
+
+		return new Range( start, end );
+	}
+
+	_getTransformedByMergeOperation( operation ) {
+		const start = this.start._getTransformedByMergeOperation( operation );
+		const end = this.end._getTransformedByMergeOperation( operation );
+
+		if ( start.isAfter( end ) ) {
+			// This happens in the following case:
 			//
-			// In all examples `[]` is `this` and `{}` is `sourceRange`, while `^` is move target position.
+			// <p>abc</p>{<p>x}yz</p> -> <p>abcx}yz</p>{
+			//                           <p>abcx{yz</p>}
 			//
-			// Example:
-			// <p>xx</p>^<w>{<p>a[b</p>}</w><p>c]d</p>   -->   <p>xx</p><p>a[b</p><w></w><p>c]d</p>
-			// ^<p>xx</p><w>{<p>a[b</p>}</w><p>c]d</p>   -->   <p>a[b</p><p>xx</p><w></w><p>c]d</p>  // Note <p>xx</p> inclusion.
-			// <w>{<p>a[b</p>}</w>^<p>c]d</p>            -->   <w></w><p>a[b</p><p>c]d</p>
-			if (
-				( sourceRange.containsPosition( this.start ) || sourceRange.start.isEqual( this.start ) ) &&
-				this.containsPosition( sourceRange.end ) &&
-				this.end.isAfter( targetPosition )
-			) {
-				const start = this.start._getCombined(
-					sourcePosition,
-					targetPosition._getTransformedByDeletion( sourcePosition, howMany )
-				);
-				const end = this.end._getTransformedByMove( sourcePosition, targetPosition, howMany, false, false );
+			return new Range( end, start );
```

Comment:
There was `x` in the range before merge operation and `yz` in the range after the merge. Does not look like the correct result.